src/main/java/com/zendesk/maxwell/RowMap.java

```diff
@@ -215,7 +217,7 @@ public String toJSON() throws IOException {
 			}
 		}
 
-		writeMapToJSON("data", this.data, false);
+		writeMapToJSON("data", this.data, !this.omitNulls);
 
 		if ( !this.oldData.isEmpty() ) {
 			writeMapToJSON("old", this.oldData, true);
```

Comment:
that's intentional: `oldData` only contains data that's changed in the row, and we *always* put a null value in there if the row was null.